2016-10-24  Katrina Alexis Wijaya  <wijaya@lnxsrv07.seas.ucla.edu>

        * timespec.h (timespec_cmp): Modified function so there is no risk
        of overflow. Now uses conditions to determine return value.
---
--- lib/timespec_original.h	2016-10-24 23:51:47.345131000 -0700
+++ lib/timespec.h	2016-10-24 23:54:19.907317000 -0700
@@ -45,8 +45,35 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  int cmp = 0;
+
+  if (a.tv_sec < b.tv_sec)
+    {
+      cmp = -1;
+    }
+  else if (a.tv_sec > b.tv_sec)
+    {
+      cmp = 1;
+    }
+
+  if (cmp != 0)
+    {
+      return cmp;
+    }
+  else
+    {
+      if (a.tv_nsec < b.tv_nsec)
+	{
+	  cmp = -1;
+	}   
+      else if (a.tv_nsec > b.tv_nsec)
+	{
+	  cmp = 1;
+	}
+    }
+
+  return cmp;
+
 }
 
 # if ! HAVE_DECL_NANOSLEEP
